Assets/Mirror/Editor/Weaver/Readers.cs

```diff
@@ -49,8 +50,72 @@ public static void Init(AssemblyDefinition CurrentAssembly)
                 { Weaver.transformType.FullName, Resolvers.ResolveMethod(networkReaderType, CurrentAssembly, "ReadTransform") },
                 { "System.Byte[]", Resolvers.ResolveMethod(networkReaderType, CurrentAssembly, "ReadBytesAndSize") },
             };
+
+            customReaders = new Dictionary<string, MethodReference>();
+            RegisterCustomReaders(CurrentAssembly);
+        }
+
+        static void RegisterCustomReaders(AssemblyDefinition currentAssembly)
+        {
+            foreach (TypeDefinition td in currentAssembly.MainModule.Types)
+            {
+                if (td.IsClass && td.BaseType.CanBeResolved())
+                {
+                    RegisterCustomReaders(td);
+                }
+            }
+        }
+
+        static void RegisterCustomReaders(TypeDefinition td)
+        {
+            foreach (MethodDefinition md in td.Methods)
+            {
+                foreach (CustomAttribute ca in md.CustomAttributes)
+                {
+                    if (ca.AttributeType.FullName == Weaver.ReaderType.FullName)
+                    {
+                        RegisterCustomReader(md);
+                        break;
+                    }
+                }
+            }
+
+            foreach (TypeDefinition embedded in td.NestedTypes)
+            {
+                RegisterCustomReaders(embedded);
+            }
         }
 
+        private static void RegisterCustomReader(MethodDefinition md)
+        {
+            if (md.Parameters.Count != 1)
+            {
+                Weaver.Error($"Reader {md.FullName} must receive a NetworkReader");
```

Comment:
Since this doesnt allow more than one parameter, how about `"Reader {md.FullName} must have exactly one NetworkReader parameter and no others."`